src/components/modals/Send/steps/01-step-amount.js

```diff
@@ -171,28 +173,38 @@ export class StepAmountFooter extends PureComponent<
 
       const amount = bridge.getTransactionAmount(mainAccount, transaction)
       const useAllAmount = bridge.getTransactionExtra(mainAccount, transaction, 'useAllAmount')
+      let maxAmount
+      if (useAllAmount) maxAmount = await bridge.getMaxAmount(mainAccount, transaction)
+
       const canNext =
-        (!amount.isZero() || useAllAmount) &&
+        (!amount.isZero() || (useAllAmount && maxAmount && !maxAmount.isZero())) &&
         isRecipientValid &&
         !!isValidTransaction &&
         totalSpent.gt(0)
-      this.setState({ totalSpent, canNext, isSyncing: false })
+      this.setState({ totalSpent, maxAmount, canNext, isSyncing: false })
     } catch (err) {
       logger.critical(err)
       this.setState({ totalSpent: BigNumber(0), canNext: false, isSyncing: false })
     }
   }
 
   onNext = async () => {
-    const { totalSpent } = this.state
+    const { totalSpent, maxAmount } = this.state
     const { transitionTo, account, parentAccount, transaction } = this.props
     if (account && transaction) {
       if (
         !parentAccount &&
-        totalSpent
-          .minus(transaction.amount)
-          .times(10)
-          .gt(transaction.amount)
+        ((transaction.amount.gt(0) &&
+          totalSpent
+            .minus(transaction.amount)
+            .times(10)
+            .gt(transaction.amount)) ||
+          (maxAmount &&
+            maxAmount.gt(0) &&
+            totalSpent
+              .minus(maxAmount)
+              .times(10)
+              .gt(maxAmount)))
```

Comment:
sorry to not ask that before, but could you make that whole logic into a function that we will put in live-common? we can only later refactor that part but at least so it's getting prepared there as well..

Future thought: what if it actually would be part of the bridge into a helper function like `async getTransactionStatus(account, transaction): { expensiveFees: bool, fees: BigNumber, amount: BigNumber }` I'm sure we could put all stuff we need into one big status object like this so we can share some code too 🤔  

Asking that because we will need same for mobile soon..